samples/CMakeLists.txt

```diff
@@ -14,6 +14,10 @@ add_subdirectory(cpp)
 add_subdirectory(gpu)
 add_subdirectory(tapi)
 
+if(HAVE_OPENCL)
+  add_subdirectory(opencl)
```

Comment:
I suggest to support only standalone samples build, and don't garbage OpenCL vars for OpenCV build.